packages/material-ui-lab/src/TreeItem/TreeItem.js

```diff
@@ -263,33 +263,17 @@ const TreeItem = React.forwardRef(function TreeItem(props, ref) {
     ariaSelected = true;
   }
 
-  React.useEffect(() => {
-    if (nodeRef && treeId) {
-      const handleFocusIn = (event) => {
-        event.preventDefault();
-        const tree = ownerDocument(nodeRef).getElementById(treeId);
-
-        // Some browsers don't focus the tree when using active-descendant.
-        // Probably can remove when we drop IE11 support.
-        if (ownerDocument(nodeRef).activeElement !== tree) {
-          tree.focus();
-        }
-
-        const unfocusable = !disabledItemsFocusable && disabled;
-        if (!focused && event.currentTarget === event.target && !unfocusable) {
-          focus(event, nodeId);
-        }
-      };
-
-      // Using focusin to avoid blurring the tree.
-      nodeRef.addEventListener('focusin', handleFocusIn);
+  function handleFocus(event) {
+    // DOM focus stays on the tree which manages focus with aria-activedescendant
+    if (event.target === event.currentTarget) {
+      ownerDocument(event.target).getElementById(treeId).focus();
```

Comment:
Just realized that I no longer guard against falsy `treeId`.  What was the intention behind that @joshwooding ?

Reply:
And `if (ownerDocument(nodeRef).activeElement !== tree)` guarded against `tree` being `null`. I'm not so sure anymore if we should crash or issue a console error (which might be missed).

Reply:
@eps1lon If the context isn't there it would error but probably not worth guarding against. I think it's fine if it crashed, I didn't intend to guard against a null tree.